conf/stjude.config

```diff
@@ -0,0 +1,105 @@
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Nextflow config file for St. Jude Children's Research Hospital's High Performance Research Cluster (HPCF)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Author: Haidong Yi
+Mail: hyi@stjude.org
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+params {
+    config_profile_contact = "Haidong Yi (hyi@stjude.org)"
+    config_profile_description = "St. Jude Children's Research Hospital HPC cluster (HPCF) profile"
+    config_profile_url = "https://www.stjude.org/"
+
+    max_cpus   = 32
+    max_memory = 1024.GB
+    max_time   = 240.h
+}
+
+process {
+    // use `resourceLimits` directive to control resource
+    // Note: `resourceLimits` is a new feature starting from version 24.04.0.
+    // To use it, `Nextflow` should be upgraded to 24.04.0.
+    // see: https://github.com/nf-core/tools/issues/2923
+    resourceLimits = [
+        memory: 1024.GB,
+        cpus: 32,
+        time: 240.h
+    ]
+    executor = 'lsf'
+    scratch = false
+    cache = 'lenient'
+
+    maxRetries = 3
+    errorStrategy = {task.exitStatus in [143,137,134,139,140] ? 'retry' : 'finish'}
+
+    afterScript = 'sleep 10' // to avoid fail when using storeDir for missing output
+    beforeScript =
+    """
+    module load singularity/4.1.1
+    export SINGULARITY_TMPDIR="\$TMPDIR" # The 'TMPDIR' environment variable will be set up by LSF after a job is submitted.
+    """
+
+    // queue selection based on task configs
+    // if urgent, change default queue from 'standard' to 'priority'
+    queue = {
+        if ( task.accelerator ) {
+            'gpu'
+        } else if ( task.time < 30.min ) {
+            "short"
+        } else if ( task.memory > 512.GB ) {
+            "large_mem"
+        } else {
+            "standard"
+        }
+    }
+
+    // clusterOptions for gpu task:
+    // NOTE: gpu jobs are dispatched to HPCF's `gpu` queue
+    //
+    // To use GPU for your nextflow process:
+    // 1. GPU processes need to set the `accelerator` option in their directives. This can be configured by:
+    //   1. Adding directives in each process specifically
+    //   2. Using `process {}`-based label selectors (e.g. label 'process_gpu')
+    //
+    // 2. (optional) If you want to use docker (unsupported) or singularity (supported) on HPCF, you need to add the following container options:
+    //   containerOptions = {
+    //       workflow.containerEngine == "singularity" ? '--nv':
+    //           ( workflow.containerEngine == "docker" ? '--gpus all': null )
+    //   }
```

Comment:
For me this should be more in the README for stjude than as comment in the config file

Reply:
Hi @maxulysse, thank you for the comments. I have moved the long comments from the config file to readme: `stjude.md`.